state-sync/storage-service/server/src/handler.rs

```diff
@@ -255,14 +255,16 @@ impl<T: StorageReaderInterface> Handler<T> {
                     .request(&request)
                 );
             );
-        }
+            metrics::OPTIMISTIC_FETCH_EVENTS.with_label_values(&[peer_network_id.network_id().as_str(), "collision"]).inc();
```

Comment:
These could be in a separate PR 😄 Also, not entirely sure about the value of the metric, e.g., we expect this to happen, but I'm not sure if the frequency means much.